### PR TITLE
Implement Mapr Installation

### PR DESCRIPTION
**Pull request by sun77**
Including recipes to :
  - Configuring repositories
  - Install prerequisite package
  - Install Mapr/Hadoop/Yarn component

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+### Main attributes ###
+default['mapr']['cluster_name'] = 'default-cluster'
+default['mapr']['release']['version'] = '6.0.1'
+default['mapr']['ecosystem']['version'] = '5.0'
+
+default['mapr']['user'] = 'mapr'
+default['mapr']['uid'] = 628
```
Comment (jmauro) [inline attributes/default.rb]: Why do you need a specific id?

File: attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+default['mapr']['group'] = 'mapr'
+default['mapr']['gid'] = 628
+
+### Repositories attributes ###
+default['mapr']['repositories']['base_url'] = 'http://package.mapr.com/releases/'
+
+default['mapr']['repositories']['release_url'] = ::File.join(
+  default['mapr']['repositories']['base_url'],
```
Comment (jmauro) [inline attributes/default.rb]: Shouldn't it be `node` instead of `default`

File: attributes/default.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  "v#{default['mapr']['release']['version']}",
```
Comment (jmauro) [inline attributes/default.rb]: The same

File: attributes/default.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  'redhat',
+)
+
+default['mapr']['repositories']['gpg_url'] = "#{default['mapr']['repositories']['base_url']}/pub/maprgpg.key"
```
Comment (jmauro) [inline attributes/default.rb]: The same

File: attributes/default.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  'redhat',
+)
+
+### Java Attributes ###
+default['java']['install_flavor'] = 'openjdk'
+default['java']['jdk_version'] = '8'
+
+### Prerequisites ###
+default['mapr']['prerequisite']['packages'] = %w[
+  mc
```
Comment (jmauro) [inline attributes/default.rb]: This package is mandatory for mapr?

File: recipes/compute.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#
+# Cookbook Name:: mapr
+# Recipe:: compute
+#
+# Copyright:: 2018, The Authors, All Rights Reserved.
+
+node['mapr']['compute']['packages'].each do |p|
+  package p
```
Comment (jmauro) [inline recipes/compute.rb]: You can directly pass an array.

File: recipes/core.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#
+# Cookbook Name:: mapr
+# Recipe:: core
+#
+# Copyright:: 2018, The Authors, All Rights Reserved.
+
+# Install mapr-core
+node['mapr']['core']['packages'].each do |p|
+  package p
```
Comment (ashangit) [inline recipes/core.rb]: same

File: recipes/nfs.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+#
+# Cookbook Name:: mapr
+# Recipe:: default
+#
+# Copyright:: 2018, The Authors, All Rights Reserved.
+
+node['mapr']['nfs']['packages'].each do |p|
+  package p
```
Comment (ashangit) [inline recipes/nfs.rb]: same

File: recipes/prerequisites.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+# Configure repositories
+include_recipe 'mapr::repositories'
+
+# Install java
+include_recipe 'java::default'
+
+# Install required packages
+node['mapr']['prerequisite']['packages'].each do |p|
+  package p
```
Comment (ashangit) [inline recipes/prerequisites.rb]: same

**Comment by ashangit** [comment]
Perhaps adding run of kitchen test in travis will spot those issue during checks

File: attributes/default.rb (hunk 1 above)
Comment (sun77) [reply attributes/default.rb]: It's from MapR documentation : 
    ```The “MapR user” is the user that MapR services run as (typically named mapr or hadoop) on each node. The MapR user has full privileges to administer the cluster. Administrative privilege with varying levels of control can be assigned to other users as well.

Before installing MapR, decide on the name, user id (UID) and group id (GID) for the MapR user. The MapR user must exist on each node, and the user name, UID and primary GID must match on all nodes.```

Comment (jmauro) [reply attributes/default.rb]: How could guarantee then the uid is not already on all nodes?

**Comment by ashangit** [comment]
LGTM

File: attributes/default.rb (hunk 1 above)
Comment (sun77) [reply attributes/default.rb]: I've removed this . I will ask to MapR why it's needed and put it back if really needed.

File: recipes/zookeeper.rb [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+# Cookbook Name:: mapr
+# Recipe:: zookeeper
+#
+# Copyright:: 2018, The Authors, All Rights Reserved.
+
+# Create and change owner for /opt/mapr/zkdata
+directory node['mapr']['zookeeper']['datadir'] do
+  owner node['mapr']['user']
+  group node['mapr']['group']
+  action :create
```
Comment (jmauro) [inline recipes/zookeeper.rb]: No `recursive`?

File: metadata.rb [hunk 11]
```diff
@@ -1,10 +1,15 @@
 name             'mapr'
-maintainer       'The Authors'
-maintainer_email 'you@example.com'
-license          'All rights reserved'
+maintainer       'Criteo'
+maintainer_email 'use_github_issues@criteo.com'
```
Comment (jmauro) [inline metadata.rb]: Couldn't you  provide a valid email?

**Pull request merged**